File: software/examples/c/example_simple.c
#include <stdio.h>

#include "ip_connection.h"
#include "bricklet_dust_detector.h"

#define HOST "localhost"
#define PORT 4223
#define UID "XYZ" // Change to your UID

int main(void) {
	// Create IP connection
	IPConnection ipcon;
	ipcon_create(&ipcon);

	// Create device object
	DustDetector dd;
	dust_detector_create(&dd, UID, &ipcon);

	// Connect to brickd
	if(ipcon_connect(&ipcon, HOST, PORT) < 0) {
		fprintf(stderr, "Could not connect\n");
		return 1;
	}
	// Don't use device before ipcon is connected

	// Get current dust density (unit is µg/m³)
	uint16_t dust_density;
	if(dust_detector_get_dust_density(&dd, &dust_density) < 0) {
		fprintf(stderr, "Could not get dust density, probably timeout\n");
		return 1;
	}

	printf("Dust Density: %d µg/m³\n", dust_density);

	printf("Press key to exit\n");
	getchar();
	ipcon_destroy(&ipcon); // Calls ipcon_disconnect internally
	return 0;
}

File: software/examples/go/example_simple.go
package main

import (
	"fmt"
	"github.com/tinkerforge/go-api-bindings/dust_detector_bricklet"
	"github.com/tinkerforge/go-api-bindings/ipconnection"
)

const ADDR string = "localhost:4223"
const UID string = "XYZ" // Change XYZ to the UID of your Dust Detector Bricklet.

func main() {
	ipcon := ipconnection.New()
	defer ipcon.Close()
	dd, _ := dust_detector_bricklet.New(UID, &ipcon) // Create device object.

	ipcon.Connect(ADDR) // Connect to brickd.
	defer ipcon.Disconnect()
	// Don't use device before ipcon is connected.

	// Get current dust density.
	dustDensity, _ := dd.GetDustDensity()
	fmt.Printf("Dust Density:  µg/m³\n", dustDensity)

	fmt.Print("Press enter to exit.")
	fmt.Scanln()

}
